docker-compose.dev.yml

```diff
@@ -1,27 +1,82 @@
 services:
+  nginx:
+    image: nginx:latest
+    ports:
+      - 8080:8080
+    configs:
+      - source: nginx_config
+        target: /etc/nginx/nginx.conf
+    networks:
+      - public
+    depends_on:
+      server:
+        condition: service_healthy
+        restart: true
+
   server:
     build:
       context: ./server
       dockerfile: Dockerfile.dev
-    ports:
-      - '5001:5001'
+    # ports:
+    #   - 5001:5001
     volumes:
       - ./server:/app
       - /app/node_modules
     environment:
       - NODE_ENV=development
     env_file:
       - .env
+    networks:
+      - private
+      - public
+    depends_on:
+      db:
+        condition: service_healthy
+        restart: true
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://server:5001"]
+      interval: 5s
+      retries: 5
+      start_period: 5s
+      timeout: 5s
 
   db:
     image: postgres:latest
-    ports:
-      - '5432:5432'
-    env_file:
-      - .env
+    # ports:
+    #   - "5432:5432"
+    environment:
```

Comment:
we don't need to set environment here
environments are loaded from the env_file option loading from the `.env` file

Reply:
mentioned this in below comment